### DISCUSSION TITLE
How to access coefficients for linear models?

### DISCUSSION THREAD

**Discussion by rokopi-byte** [autogluon/autogluon#4591] [Q&A]
I'm using a TabularPredictor and fitting it with `hyperparameters={'LR': {'penalty': 'L2'}}`. Now, I'm looking to retrieve the model coefficients. The only way I've found to access them is by using` predictor._trainer.load_model('LinearModel').model.coef_`. However, this approach returns 9 coefficients, while I only have 4 features. 

I understand that the discrepancy is likely due to one-hot encoding, as I have 2 continuous features and 2 categorical features (one with 3 unique values and the other with 4), resulting in a total of 9 features after encoding.

Since feature names are not directly stored in the model, I'm unsure how to map each `coef_` value back to its corresponding feature. Can I reliably use the order provided by `predictor._trainer.load_model('LinearModel').features` to match coefficients to features, taking into account the one-hot encoding?

For example, if `predictor._trainer.load_model('LinearModel').features = ['a', 'b', 'c', 'd']`, where 'a' and 'b' are continuous features, 'c' is a categorical feature with 3 possible values, and 'd' is a categorical feature with 4 possible values, then `predictor._trainer.load_model('LinearModel').model.coef_` returns `[coef_a, coef_b, coef_c_1, coef_c_2, coef_c_3, coef_d_1, coef_d_2, coef_d_3, coef_d_4]`. In this example, `coef_c_1` represents the coefficient for the first one-hot encoded value of the 'c' feature.

Is this mapping approach correct, or is there a more accurate way to retrieve the coefficients with feature associations?

**Comment by barstow2** [comment]
I'm wondering this too.  I think you're correct.  In my situation, the coefficients seem to line up with the feature importance.
